go-controller/pkg/util/multi_network.go

```diff
@@ -761,6 +761,61 @@ func ValidateNetConf(nadName string, netconf *ovncnitypes.NetConf) error {
 		return fmt.Errorf("the subnet attribute must be defined for layer2 primary user defined networks")
 	}
 
+	if netconf.Topology != types.LocalnetTopology && netconf.Name != types.DefaultNetworkName {
```

Comment:
should we also check for `netconf.Role == types.NetworkRolePrimary`  before doing `subnetOverlapCheck` ?

Reply:
https://github.com/openshift/enhancements/pull/1623/files#diff-76f8949dbcefb73686f569b18267b152f0f1ef3d71ce9888046aa93f97566005R292 makes me think that we need to run subnet overlap check regardless of netconf.Role.